BASE/Test/Microsoft.ApplicationInsights.Test/Microsoft.ApplicationInsights.Tests/Channel/TransmissionTest.cs

```diff
@@ -501,6 +501,31 @@ public async Task SendAsyncLogsIngestionReponseTimeAndStatusCode()
                     }
                 }
             }
+
+            [TestMethod]
+            public async Task TestTransmissionStatusEventHandler()
+            {
+                var handler = new HandlerForFakeHttpClient
+                {
+                    InnerHandler = new HttpClientHandler(),
+                    OnSendAsync = (req, cancellationToken) =>
+                    {
+                        return Task.FromResult<HttpResponseMessage>(new HttpResponseMessage());
+                    }
+                };
+
+                using (var fakeHttpClient = new HttpClient(handler))
+                {
+                    // Instantiate Transmission with the mock HttpClient                  
+                    Transmission transmission = new Transmission(testUri, new byte[] { 1, 2, 3, 4, 5 }, fakeHttpClient, string.Empty, string.Empty);
```

Comment:
make a test with actual ITelemetry items being sent, each with separate ikey. Validate that the TransmissionStatusEvent receives the Transmission, and it can read the individual ikeys.

We probably need to give a wrapper to help with 206/partial success status codes. If a few items from batch is rejected, the TransmissionStatusEvent should be able to tell which ones were accepted/rejected. This requires knowledge about ingestion response format, which this SDK knows - but not the end user. So a helper/wrapper will help here.

Reply:
TransmissionStatusEvent has response from ingestion endpoint and not the actual transmission. **Sender** parameter in the eventhandler method is the transmission object.

Added a new test case for this scenario. 
The sample in our todo list will cover this complete flow.